src/lib/signature_kind/mina_signature_kind.mli

```diff
@@ -3,6 +3,20 @@ type t = Mina_signature_kind_type.t =
   | Mainnet
   | Other_network of string
 
+val bin_t : t Bin_prot.Type_class.t
```

Comment:
Did you try `type t = ... [@@deriving bin_io]`?

`bin_io_unversioned` is implementation-only deriver, but just `bin_io` normally works